cdptools/research_utils/transcripts.py

```diff
@@ -1,72 +1,154 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from concurrent.futures import ThreadPoolExecutor
+from functools import partial
 from pathlib import Path
-from typing import Dict, Optional, Union
+from typing import Any, Dict, NamedTuple, Optional, Union
 
 import pandas as pd
 
-from cdptools.databases import Database
+from cdptools.databases import Database, OrderOperators
 from cdptools.file_stores import FileStore
 
+###############################################################################
+
+ALLOWED_ORDER_BY_FIELDS = [
+    "created",
+    "confidence"
+]
+
+###############################################################################
+
+
+class TrancriptJoin(NamedTuple):
+    event_id: str
+    transcript_details: Optional[Dict[str, Any]]
+
+
+def _get_transcript_for_event(event_id: str, db: Database, order_by_field: str) -> Dict[str, Any]:
+    # Get the most recent or highest confidence transcript for the event
+    results = db.select_rows_as_list(
+        table="transcript",
+        filters=[
+            ("event_id", event_id)
+        ],
+        order_by=(order_by_field, OrderOperators.desc),
+        limit=1
+    )
+
+    # Return result if found
+    if len(results) == 1:
+        return TrancriptJoin(event_id=event_id, transcript_details=results[0])
+
+    # Return none to be filtered out
+    return TrancriptJoin(event_id=event_id, transcript_details=None)
 
-def get_most_recent_transcript_manifest(db: Database) -> pd.DataFrame:
+
+def _get_file(file_id: str, db: Database) -> Dict[str, Any]:
+    return db.select_row_by_id(table="file", id=file_id)
+
+
+def get_transcript_manifest(db: Database, order_by_field: str = "confidence") -> pd.DataFrame:
```

Comment:
This is great, I didn't know much about dataframes before but they are pretty cool and I learned a lot from this!